community/values/src/main/java/org/neo4j/values/storable/DateTimeValue.java

```diff
@@ -438,8 +438,7 @@ public boolean equals( Value other )
         {
             ZonedDateTime that = ((DateTimeValue) other).value;
             return value.toLocalDateTime().equals( that.toLocalDateTime() ) &&
-                   value.getOffset().equals( that.getOffset() ) &&
-                   !areDifferentZones( value.getZone(), that.getZone() );
+                    areSameZone( value.getZone(), that.getZone() );
```

Comment:
This looks neater, but I can't see the behavioural change

Reply:
The behavioral change was not here, but instead `areDifferentZone` was broken. I decided to code it as `areSameZone` instead, which seemed easier and more intuitive.